### PR TITLE
Added option group in Select component

### PR DESCRIPTION
**Pull request by jaieds**
### Screenshots

- https://d.pr/v/twIyTg

### Checklist:

-   [ ] My code is tested
-   [ ] My code passes the PHPCS tests
-   [ ] I've created the npm build.
-   [ ] My code follows accessibility standards 
-   [ ] My code has proper inline documentation 
-   [ ] I've included any necessary tests 
-   [ ] I've included developer documentation 
-   [ ] I've added proper labels to this pull request

### CODE REVIEW

**Review by vrundakansara — APPROVED** [review]

File: changelog.txt [hunk 1]
```diff
@@ -1,19 +1,23 @@
+Version 1.4.0 - 26th December, 2024
```
Comment (vrundakansara) [inline changelog.txt:1]: This should be v1.3.4 since no new component is introduced. @jaieds

**Review by vrundakansara — CHANGES_REQUESTED** [review]

File: changelog.txt (hunk 1 above)
Comment (jaieds) [reply changelog.txt:1]: Okay. I will update it to `v1.3.4`.

File: src/components/select/select.tsx [hunk 2]
```diff
@@ -325,43 +386,128 @@ export function SelectOptions( {
...
+					if ( searchKeyword ) {
+						const valueProp = groupChild.props.value;
+						if ( typeof valueProp === 'object' ) {
+							return valueProp[ searchBy ]
+								.toLowerCase()
+								.includes( searchKeyword.toLowerCase() );
+						}
+						return valueProp
+							.toLowerCase()
+							.includes( searchKeyword.toLowerCase() );
```
Comment (ravindra114) [inline src/components/select/select.tsx:412]: we can add a common utility function here.

Comment (jaieds) [reply src/components/select/select.tsx:412]: Both conditions are different and used for different purposes.

**Review by vrundakansara — APPROVED** [review]

**Pull request merged**